app/domain/authentication/authn_jwt/signing_key/public_signing_keys.rb

```diff
@@ -0,0 +1,43 @@
+module Authentication
+  module AuthnJwt
+    module SigningKey
+      # This class is a POJO class presents public-keys structure
+      class PublicSigningKeys
+        include ActiveModel::Validations
+        include AttrRequired
+
+        VALID_TYPES =  %w[jwks].freeze
+
+        attr_required(:type, :value)
+
+        validates(*required_attributes, presence: true)
+        validates(:type, inclusion: { in: VALID_TYPES, message: "'%{value}' is not a valid public-keys type" })
+        validate(:validate_value_is_jwks, if: -> { @type == "jwks" })
+
+        def initialize(hash)
+          raise Errors::Authentication::AuthnJwt::InvalidPublicKeys.new("the value is not in valid JSON format") unless
+            hash.is_a?(Hash)
+
+          hash = hash.with_indifferent_access
+          required_attributes.each do |key|
+            send("#{key}=", hash[key])
+          end
+        end
+
+        def validate!
+          raise Errors::Authentication::AuthnJwt::InvalidPublicKeys.new(errors.full_messages.to_sentence) unless valid?
+        end
+
+        private
+
+        def validate_value_is_jwks
+          errors.add(:value, "is not a valid JWKS (RFC7517)") unless
```

Comment:
We have several error messages for key validation , this one is an addition ? the RFC is not to global (JSON Web Key (JWK))?

Reply:
The validations only for public-keys use case where the value is came from customer and is not fetched directly from keys provider